compiler/noirc_evaluator/src/ssa/plonky2_gen/mod.rs

```diff
@@ -294,24 +297,34 @@ impl P2Target {
     }
 }
 
-pub(crate) struct Builder {
-    builder: P2Builder,
+pub(crate) struct Builder<TAsmWriter: AsmWriter> {
```

Comment:
Is it necessary for the Builder to be parametrized?

Reply:
If we use a trait with compile-time dispatch, yes. Alternatively, we can use a trait with dynamic dispatch, which uses a VMT table, but that is slightly slower and needs some nontrivial fighting with Rust's borrow checker in order to deal with the object's lifetimes. The third option is not to use a trait at all.

Reply:
I was thinking about dynamic dispatch, I guess, but there's no need to fight the borrow checker, if it's resisting :)